proposals/Level-1.md

```diff
@@ -78,94 +77,69 @@ instruction. Thrown exceptions are handled as follows:
-appropriate event handler. The event handler is then run. That event handler
-can either resume control at the point of the event handler, or optionally
-resume by sending values back to the yielding instruction allow the
-originating code to resume.
+An event handler allows one to process an event generated by a block of
+code. Events yield the current execution and look for a corresponding event
+handler. If found, the corresponding event handler is run.  The event handler
+can optionally resume by sending values back to the yielded instruction,
+allowing the originating code to resume.
 
 Exceptions are a special case of an event in that they never resume. Similarly,
-a throw instruction is the yielding instruction of an exception, and the
-corresponding catching block is the event handler of the exception.
+a throw instruction is the yielding event of an exception. The catch block
+associated with a try block defines how to handle the throw.
 
-An `event` is an internal construct in WebAssembly that is maintained by the
-embedder. WebAssembly events are defined by a new `event section` of a
-WebAssembly module.
+An `event` (such as an exception) is an internal construct in WebAssembly that
+is maintained by embedder. WebAssembly events are defined by a new `event`
+section of a WebAssembly module. The event section is a list of event types
+that are defined by the module.
 
-The event section defines a list of event types that may be handled by the
-module.  Each `event type` has (at a minimum) an `event specifier` and a `type
-signature`. The event specifier defines the kind of event while the type
-signature defines the list of values associated with the event.  Depending on
+Each `event type` has a `event specifier` and a `type signature`. Depending on
 the value of the event specifier, additional values may be specified in the
-event type.
 
 For level 1, the only event specifier is the constant 0, and denotes that the
-event is an `exception`. It has no additional values specified (other than the
-type signature). Future extensions my add additional event specifiers, such as
-resumable events.
-
-The `type signature` of an event type is an index into the `type` section.  The
-type signature is defined by a function type. The parameter list of the type
-signature defines the list of values returned to the corresponding enclosing
-catch. The result type defines the list of values to be used when resuming on an
-event. For exceptions, that can't resume, the result type must always be type
-'void'.
-
-Note: Resumable events may not use the `throw` instruction. Rather, they may use
-a new (yet to be defined) `yeild` instruction that passes value(s) to its
-enclosing contexts. Such an instruction would allow the calling context to
-return values back, and resume with the instruction following the yeild
-instruction, based on the values returned.
+event type corresponds to an exception. Future extensions my add additional
+event specifiers.
```

Comment:
Is it better to be more vague about event specifiers and just say that event entries are required to have a 0 byte, which is reserved for future expansion?